pkg/apis/common/v1/common.go

```diff
@@ -49,6 +49,9 @@ type ObjectSelector struct {
 	Name string `json:"name"`
 	// Namespace of the Kubernetes object. If empty, defaults to the current namespace.
 	Namespace string `json:"namespace,omitempty"`
+	// ServiceName is the name of an existing Kubernetes service which will be used to make requests to the referenced
+	// object. If left empty the default HTTP service of the referenced resource will be used.
+	ServiceName string `json:"serviceName,omitempty"`
```

Comment:
Let's make it clear in the comment and generated doc that the service must exist in the referenced object namespace?